protocols/peer/contracts/Peer.sol

```diff
@@ -50,10 +56,59 @@ contract Peer is IPeer, Ownable {
   ) public {
     swapContract = ISwap(_swapContract);
     if (_peerContractOwner != address(0)) {
-      transferOwnership(_peerContractOwner);
+      whitelist[_peerContractOwner] = true;
+      super.transferOwnership(_peerContractOwner);
     }
   }
 
+  /**
+    * @notice determines if an address to interact with this peer
+    * @param addressToCheck the address to check if whitelisted or not 
+    */
+  function isWhitelisted(address addressToCheck) external returns(bool) {
+    return whitelist[addressToCheck];
+  }
+
+  /**
+    * @notice adds to the list of whitelisted accounts that can interact with this peer
+    * @dev only callable by the owner of the contract
+    * @param addressToAdd the address to add to the whitelist
+    */
+  function addToWhitelist(address addressToAdd) external onlyOwner {
+    whitelist[addressToAdd] = true;
+    emit WhitelistAdded(addressToAdd);
+  }
+
+  /**
+    * @notice removes from the list of whitelisted accounts that can interact with this peer
+    * @dev only callable by the owner of the contract
+    * @param addressToRemove the address to add to the whitelist
+    */
+  function removeFromWhitelist(address addressToRemove) external onlyOwner {
+    require(addressToRemove != owner(), "OWNER_MUST_BE_WHITELISTED");
+    whitelist[addressToRemove] = false;
```

Comment:
reason for choosing setting to false versus del whitelist[addressToRemove]

Reply:
no reason. good catch.